<a href="https://colab.research.google.com/github/JSEFERINO/C_Datos/blob/main/8_C_Datos_DataFrame.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Curso Introductorio a la Ciencia de Datos por Julio Hurtado Márquez**
## **Parte I**

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

In [ ]:
import pandas as pd
import numpy as np

#**34. Creando un dataFrame usando arrays**
##**a. Construyamos una matriz que contenga el día y fecha de cumpleaños de mi familia**

In [ ]:
# Creando un array
data = np.array([[2,6], [9,25], [3,21], [7,21], [12,10], [9,21]])
data

array([[ 2,  6],
       [ 9, 25],
       [ 3, 21],
       [ 7, 21],
       [12, 10],
       [ 9, 21]])

## **b. Creamos el DataFrame**

In [ ]:
#Creando un data Frame
pd.DataFrame(data)

0   1
0   2   6
1   9  25
2   3  21
3   7  21
4  12  10
5   9  21

## **c. Las Filas son los membros de la Familia y el mes de cumpleaños**

In [ ]:
#Creando un data Frame
fh = pd.DataFrame(data, index = ["Julio", "Marelis", "Aris", "Dayana", "Cofy", "Chiqui"], columns= ["Mes","Dia"])
fh

Mes  Dia
Julio      2    6
Marelis    9   25
Aris       3   21
Dayana     7   21
Cofy      12   10
Chiqui     9   21

#**35. Creando un DataFrame en Python usando lista**

## **a. Creamos una lista con cumpleaños**

In [ ]:
# **a. Creamos una lista con cumpleaños**
cumpleaños3 = [[2,6], [9,25], [3,21], [7,21], [12,10], [9,21]]
cumpleaños3

[[2, 6], [9, 25], [3, 21], [7, 21], [12, 10], [9, 21]]

## **b. Creando un data Frame de la misma forma que en el caso 1**

In [ ]:
## **Creando un data Frame**
fh2 = pd.DataFrame(cumpleaños3, index = ["Julio", "Marelis", "Aris", "Dayana", "Cofy", "Chiqui"], columns= ["Mes","Dia"])
fh2

Mes  Dia
Julio      2    6
Marelis    9   25
Aris       3   21
Dayana     7   21
Cofy      12   10
Chiqui     9   21

## **c. Cree un DataFrame en Python con los nombres de los departamentos y sus capitaes**

In [ ]:
# **c. Creando la lista**
costa = [["Guajira","RioHacha"], ["Magdalena","Santa Marta"], ["Cesár","Valledupar"], ["Atlántico","Barranquilla"], ["Bolivar","Cartagena"], ["Sucre","Sincelejo"], ["Córdova", "Montería"]]

In [ ]:
#Creando un data Frame
C_Caribe = pd.DataFrame(costa, index = ["region1", "region2", "region3", "region4", "region5", "region6", "region7"], columns= ["Dpto","Capital"])
C_Caribe

Dpto       Capital
region1    Guajira      RioHacha
region2  Magdalena   Santa Marta
region3      Cesár    Valledupar
region4  Atlántico  Barranquilla
region5    Bolivar     Cartagena
region6      Sucre     Sincelejo
region7    Córdova      Montería

#**35. Creando un DataFrame en Python usando diccionarios**

## **a. Creando las listas de las ciudades y dptos**

In [ ]:
ciudad_3 = ['Cartagena', 'Barranquilla', 'Santa Marta', 'Montería', 'Sincelejo', 'Valledupar', 'Riohacha']

In [ ]:
dpto4 = ['Bolívar', 'Atlántico', 'Magdalena', 'Córdoba', 'Sucre', 'Cesar', 'Guajira']

## **b. Creando los diccionarios**

In [ ]:
## **b. Creando los diccionarios**
dic_Dptos = {"Ciudad": ciudad_3, "Departamento":dpto4}

## **c. Creando el DataFrame con los diccionarios**

In [ ]:
## **c. Creando el DataFrame con los diccionarios**
dic_ciudad = pd.DataFrame(dic_Dptos)
dic_ciudad

Ciudad Departamento
0     Cartagena      Bolívar
1  Barranquilla    Atlántico
2   Santa Marta    Magdalena
3      Montería      Córdoba
4     Sincelejo        Sucre
5    Valledupar        Cesar
6      Riohacha      Guajira

#**36. Crear un DataFrame usando archivos csv**
##**a. Subiendo un archivo csv a Python**

In [ ]:
import cv2 as cv
from google.colab import files
uploaded=files.upload()

Saving DATOS2023py.csv to DATOS2023py (1).csv


## **b. Construimos el DataFrame con el archivo excel**

In [ ]:
# **b. Construimos el DataFrame con el archivo excel**
df_datos2023 = pd.read_csv("/content/DATOS2023py.csv", delimiter=';')
df_datos2023

CURSO DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  \
0    ESTADISTICA I        4,3     COMS    18     SI    55       164   
1    ESTADISTICA I        3,1     COMS    19     SI    65       165   
2    ESTADISTICA I        4,2     DERH    25     NO    56       170   
3    ESTADISTICA I        3,6     DERH    23     SI    65       170   
4    ESTADISTICA I        3,9     PSIC    19     SI    50       169   
..             ...        ...      ...   ...    ...   ...       ...   
116  ESTADISTICA I        3,5     PSIC    18     SI    60       163   
117  ESTADISTICA I        3,5     PSIC    18     SI    60       180   
118  ESTADISTICA I        2,5     COMS    21     NO    45       172   
119  ESTADISTICA I          3     COMS    20     NO    50       160   
120  ESTADISTICA I        3,5     DERH    21     NO    57       160   

     SATISFACC  SABER 11 PROMEDIO SEXO ESTRATO Herramientas  
0            4       350     4,26    F      II         PTWF  
1            5       328     4,11    F       I         CYDT  
2            5       285      3,7    M       I         CYDT  
3            5       280      4,1    F      II         PCWF  
4            5       319        4    F      II         CYWF  
..         ...       ...      ...  ...     ...          ...  
116          5       310      3,9    F     III         PTWF  
117          3       346      3,9    M      IV         PCWF  
118          5       221      3,7    F      II         CYWF  
119          4       332      3,5    F      II         CYDT  
120          4       293      3,5    F     III         PTWF  

[121 rows x 13 columns]

## **c. Renombramos el archivo para mejorarlo**

In [ ]:
df_datos = df_datos2023
df_datos

CURSO DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  \
0    ESTADISTICA I        4,3     COMS    18     SI    55       164   
1    ESTADISTICA I        3,1     COMS    19     SI    65       165   
2    ESTADISTICA I        4,2     DERH    25     NO    56       170   
3    ESTADISTICA I        3,6     DERH    23     SI    65       170   
4    ESTADISTICA I        3,9     PSIC    19     SI    50       169   
..             ...        ...      ...   ...    ...   ...       ...   
116  ESTADISTICA I        3,5     PSIC    18     SI    60       163   
117  ESTADISTICA I        3,5     PSIC    18     SI    60       180   
118  ESTADISTICA I        2,5     COMS    21     NO    45       172   
119  ESTADISTICA I          3     COMS    20     NO    50       160   
120  ESTADISTICA I        3,5     DERH    21     NO    57       160   

     SATISFACC  SABER 11 PROMEDIO SEXO ESTRATO Herramientas  
0            4       350     4,26    F      II         PTWF  
1            5       328     4,11    F       I         CYDT  
2            5       285      3,7    M       I         CYDT  
3            5       280      4,1    F      II         PCWF  
4            5       319        4    F      II         CYWF  
..         ...       ...      ...  ...     ...          ...  
116          5       310      3,9    F     III         PTWF  
117          3       346      3,9    M      IV         PCWF  
118          5       221      3,7    F      II         CYWF  
119          4       332      3,5    F      II         CYDT  
120          4       293      3,5    F     III         PTWF  

[121 rows x 13 columns]

## **d. Mostramos las primeras cinco filas del archivo**

In [ ]:
## **d. Mostramos las primeras cinco filas del archivo**
df_datos.head()

CURSO DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  SATISFACC  \
0  ESTADISTICA I        4,3     COMS    18     SI    55       164          4   
1  ESTADISTICA I        3,1     COMS    19     SI    65       165          5   
2  ESTADISTICA I        4,2     DERH    25     NO    56       170          5   
3  ESTADISTICA I        3,6     DERH    23     SI    65       170          5   
4  ESTADISTICA I        3,9     PSIC    19     SI    50       169          5   

   SABER 11 PROMEDIO SEXO ESTRATO Herramientas  
0       350     4,26    F      II         PTWF  
1       328     4,11    F       I         CYDT  
2       285      3,7    M       I         CYDT  
3       280      4,1    F      II         PCWF  
4       319        4    F      II         CYWF

## **e. Mostramos las ultimas cinco filas del DataFrame**

In [ ]:
## **e. Mostramos las ultimas cinco filas del DataFrame**
df_datos.tail()

CURSO DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  \
116  ESTADISTICA I        3,5     PSIC    18     SI    60       163   
117  ESTADISTICA I        3,5     PSIC    18     SI    60       180   
118  ESTADISTICA I        2,5     COMS    21     NO    45       172   
119  ESTADISTICA I          3     COMS    20     NO    50       160   
120  ESTADISTICA I        3,5     DERH    21     NO    57       160   

     SATISFACC  SABER 11 PROMEDIO SEXO ESTRATO Herramientas  
116          5       310      3,9    F     III         PTWF  
117          3       346      3,9    M      IV         PCWF  
118          5       221      3,7    F      II         CYWF  
119          4       332      3,5    F      II         CYDT  
120          4       293      3,5    F     III         PTWF

In [ ]:
## **f. Mostramos las primeras 10 filas del DataFrame**
df_datos.head(10)

CURSO DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  SATISFACC  \
0  ESTADISTICA I        4,3     COMS    18     SI    55       164          4   
1  ESTADISTICA I        3,1     COMS    19     SI    65       165          5   
2  ESTADISTICA I        4,2     DERH    25     NO    56       170          5   
3  ESTADISTICA I        3,6     DERH    23     SI    65       170          5   
4  ESTADISTICA I        3,9     PSIC    19     SI    50       169          5   
5  ESTADISTICA I        1,1     PSIC    19     SI    68       165          4   
6  ESTADISTICA I        3,5    CIENP    19     SI    86       184          5   
7  ESTADISTICA I        4,2    CIENP    19     NO    53       163          5   
8  PROBABILIDAD         4,4     ININ    18     NO    56       158          5   
9  PROBABILIDAD         4,1     INMT    20     SI    72       179          4   

   SABER 11 PROMEDIO SEXO ESTRATO Herramientas  
0       350     4,26    F      II         PTWF  
1       328     4,11    F       I         CYDT  
2       285      3,7    M       I         CYDT  
3       280      4,1    F      II         PCWF  
4       319        4    F      II         CYWF  
5       318      3,7    M     III         PTWF  
6       345     4,48    F      II         PTWF  
7       257      3,7    M       I         PTWF  
8       298      3,8    F      II         PTDT  
9       353      3,9    M     III         PCWF

## **g. Mostramos las últimas 10 filas del DataFrame**

In [ ]:
## **g. Mostramos las últimas 10 filas del DataFrame**
df_datos.tail(10)

CURSO DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  \
111  ESTADISTICA I        1,5     COMS    19     SI    61       156   
112  ESTADISTICA I        3,5     COMS    19     SI    66       110   
113  ESTADISTICA I        4,5     PSIC    18     SI    60       180   
114  ESTADISTICA I          4     PSIC    18     NO    47       155   
115  ESTADISTICA I        2,5     PSIC    21     SI    70       185   
116  ESTADISTICA I        3,5     PSIC    18     SI    60       163   
117  ESTADISTICA I        3,5     PSIC    18     SI    60       180   
118  ESTADISTICA I        2,5     COMS    21     NO    45       172   
119  ESTADISTICA I          3     COMS    20     NO    50       160   
120  ESTADISTICA I        3,5     DERH    21     NO    57       160   

     SATISFACC  SABER 11 PROMEDIO SEXO ESTRATO Herramientas  
111          5       238      3,5    F     III         PTWF  
112          4       255       40    M     III         PCWF  
113          3       346      3,9    M      IV         PTWF  
114          5       312     4,39    F       V         PCWF  
115          3       356      3,2    M      II         PCWF  
116          5       310      3,9    F     III         PTWF  
117          3       346      3,9    M      IV         PCWF  
118          5       221      3,7    F      II         CYWF  
119          4       332      3,5    F      II         CYDT  
120          4       293      3,5    F     III         PTWF

# **37. Atributos y Métodos del DataFrame**

#**a. Mostrar todas las filas del DataFrame**

In [ ]:
#**a. Mostrar todas las filas del DataFrame**
pd.set_option("display.max_rows",121)
df_datos

CURSO DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  \
0    ESTADISTICA I        4,3     COMS    18     SI    55       164   
1    ESTADISTICA I        3,1     COMS    19     SI    65       165   
2    ESTADISTICA I        4,2     DERH    25     NO    56       170   
3    ESTADISTICA I        3,6     DERH    23     SI    65       170   
4    ESTADISTICA I        3,9     PSIC    19     SI    50       169   
5    ESTADISTICA I        1,1     PSIC    19     SI    68       165   
6    ESTADISTICA I        3,5    CIENP    19     SI    86       184   
7    ESTADISTICA I        4,2    CIENP    19     NO    53       163   
8    PROBABILIDAD         4,4     ININ    18     NO    56       158   
9    PROBABILIDAD         4,1     INMT    20     SI    72       179   
10   PROBABILIDAD         4,8     INNV    18     SI    64       175   
11   PROBABILIDAD         4,5     ININ    17     NO    75       177   
12   PROBABILIDAD         3,9     INQU    19     SI    61       158   
13   PROBABILIDAD         3,3     ININ    17     SI    57       164   
14   PROBABILIDAD           4     INMT    18     NO    50       168   
15   ESTADISTICA I        3,8     COMS    20     SI    87       174   
16   ESTADISTICA I        3,5     COMS    16     SI    54       169   
17   ESTADISTICA I        4,2     DERH    18     SI    80       181   
18   ESTADISTICA I        3,6     DERH    19     SI    60       182   
19   ESTADISTICA I        4,1     PSIC    18     SI    90       176   
20   ESTADISTICA I        2,5     PSIC    20     SI    50       157   
21   ESTADISTICA I        3,6    CIENP    19     SI    55       156   
22   ESTADISTICA I        4,5    CIENP    18     SI    74       183   
23   PROBABILIDAD           4     INST    18     NO    63       183   
24   PROBABILIDAD           5     INAM    17     SI    48       158   
25   ESTADISTICA I        3,6     DERH    18     SI    85       179   
26   ESTADISTICA I          4     PSIC    20     SI    60       175   
27   ESTADISTICA I          3     PSIC    17     NO    61       155   
28   ESTADISTICA I          4    CIENP    18     SI    68       185   
29   PROBABILIDAD         4,5     INMT    17     SI    85       175   
30   PROBABILIDAD         4,3     INBM    17     SI    45       155   
31   ESTADISTICA I        4,5     COMS    17     NO    60       162   
32   ESTADISTICA I          3     COMS    18     SI    65       169   
33   ESTADISTICA I        3,5     DERH    21     SI    62       100   
34   ESTADISTICA I        3,2     DERH    17     NO    60       170   
35   ESTADISTICA I        3,6     PSIC    18     SI    56       163   
36   ESTADISTICA I          4     PSIC    18     SI    75       190   
37   ESTADISTICA I          5    CIENP    19     SI    73       182   
38   ESTADISTICA I        3,8    CIENP    17     NO    73       175   
39   PROBABILIDAD         4,3     INMC    24     NO    90       176   
40   PROBABILIDAD         4,3     INMC    18     SI    47       167   
41   ESTADISTICA I        4,5     COMS    19     SI    50       165   
42   ESTADISTICA I          3     COMS    17     SI    70       173   
43   ESTADISTICA I          5     DERH    20     SI    63       160   
44   PROBABILIDAD           5     INCV    17     NO    37       148   
45   PROBABILIDAD           5     FINN    18     SI    58       165   
46   PROBABILIDAD         4,8     ININ    20     SI    73       163   
47   PROBABILIDAD         2,5     INMT    17     SI    64       180   
48   PROBABILIDAD           3     INNV    18     NO    80       175   
49   PROBABILIDAD         3,8     INMC    17     NO    70       170   
50   ESTADISTICA I        4,4     COMS    18     SI    60       165   
51   ESTADISTICA I        2,5     COMS    18     SI    77       180   
52   ESTADISTICA I        4,2     DERH    17     NO    59       169   
53   ESTADISTICA I        3,7     DERH    17     SI    40       164   
54   ESTADISTICA I          4     PSIC    18     SI    49       163   
55   ESTADISTICA I          2     PSI

## **b. Accediendo al atributo shape, el cual me muestra la cantidad de filas y columnas**

In [ ]:
# **b. Accediendo al atributo shape, el cual me muestra la cantidad de filas y columnas**
df_datos.shape

(121, 13)

## **c. El Atributo index muestra la misma información**

In [ ]:
# **c. El Atributo index muestra la misma información**
df_datos.index

RangeIndex(start=0, stop=121, step=1)

## **d. El Atributo _columns_ muestra el nombre de las variables**

In [ ]:
## **d. El Atributo -columns- muestra el nombre de las variables**
df_datos.columns

Index(['CURSO', 'DEFINITIVA', 'PROGRAMA', 'EDAD', 'URBANO', 'PESO', 'ESTATURA',
       'SATISFACC', 'SABER 11', 'PROMEDIO', 'SEXO', 'ESTRATO', 'Herramientas'],
      dtype='object')

## **e. El Atributo _dtypes_ muestra el tipo de las  variables**
### **Algunas variables numéricas aparecen como categoricas, dado que les aparece ',' en lugar de '.'**

In [ ]:
## **e. El Atributo _dtypes_ muestra el tipo de las  variables**
### **Algunas variables numéricas aparecen como categoricas, dado que les aparece ',' en lugar de '.'**
df_datos.dtypes

CURSO           object
DEFINITIVA      object
PROGRAMA        object
EDAD             int64
URBANO          object
PESO             int64
ESTATURA         int64
SATISFACC        int64
SABER 11         int64
PROMEDIO        object
SEXO            object
ESTRATO         object
Herramientas    object
dtype: object

#**38. Métodos usados para DataFrame**

## **a. El Método _info()_ muestra el tipo de las  variables en un DataFrame**
### **Algunas variables numéricas aparecen como categoricas, dado que les aparece ',' en lugar de '.'**

In [ ]:
## **a. El Método _info()_ muestra el tipo de las  variables en un DataFrame**
### **Algunas variables numéricas aparecen como categoricas, dado que les aparece ',' en lugar de '.'**
df_datos.info() # La variable DEFINITIVA aparece como cadena porque usa (,)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 121 entries, 0 to 120
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   CURSO         121 non-null    object
 1   DEFINITIVA    121 non-null    object
 2   PROGRAMA      121 non-null    object
 3   EDAD          121 non-null    int64 
 4   URBANO        121 non-null    object
 5   PESO          121 non-null    int64 
 6   ESTATURA      121 non-null    int64 
 7   SATISFACC     121 non-null    int64 
 8   SABER 11      121 non-null    int64 
 9   PROMEDIO      121 non-null    object
 10  SEXO          121 non-null    object
 11  ESTRATO       120 non-null    object
 12  Herramientas  121 non-null    object
dtypes: int64(5), object(8)
memory usage: 12.4+ KB


## **b. El Método _describe()_ muestra los estadisticos  concalculado las  variables en un DataFrame**
### **Mostrando analisis descriptivos de la información del DataFrame**

In [ ]:
## **b. El Método _describe()_ muestra los estadisticos  concalculado las  variables en un DataFrame**
### **Mostrando analisis descriptivos de la información del DataFrame**
df_datos.describe() # No aparece la variable DEFINITIVA

EDAD        PESO    ESTATURA  SATISFACC    SABER 11
count  121.000000  121.000000  121.000000  121.00000  121.000000
mean    18.553719   63.239669  167.933884    4.53719  305.429752
std      1.653030   12.935755   12.369812    0.63300   43.059228
min     16.000000   37.000000  100.000000    3.00000  204.000000
25%     18.000000   55.000000  163.000000    4.00000  275.000000
50%     18.000000   61.000000  169.000000    5.00000  305.000000
75%     19.000000   73.000000  175.000000    5.00000  341.000000
max     27.000000   97.000000  192.000000    5.00000  380.000000

## **c. El Método _str.replace()_ permite reemplazar en las variables aquellas que aparecen con comas**
### **Cambiar la coma (,) por punto (.)**

In [ ]:
## **c. El Método _str.replace()_ permite reemplazar (.) en las variables aquellas que aparecen con comas (,)**
### **Cambiar la coma (,) por punto (.)**
df_datos["DEFINITIVA"]=df_datos["DEFINITIVA"].str.replace(",",".")
df_datos["PROMEDIO"]=df_datos["PROMEDIO"].str.replace(",",".")
df_datos.head()

CURSO DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  SATISFACC  \
0  ESTADISTICA I        4.3     COMS    18     SI    55       164          4   
1  ESTADISTICA I        3.1     COMS    19     SI    65       165          5   
2  ESTADISTICA I        4.2     DERH    25     NO    56       170          5   
3  ESTADISTICA I        3.6     DERH    23     SI    65       170          5   
4  ESTADISTICA I        3.9     PSIC    19     SI    50       169          5   

   SABER 11 PROMEDIO SEXO ESTRATO Herramientas  
0       350     4.26    F      II         PTWF  
1       328     4.11    F       I         CYDT  
2       285      3.7    M       I         CYDT  
3       280      4.1    F      II         PCWF  
4       319        4    F      II         CYWF

## **d. Revisamos nuevamente con _info()_ Aun la variable DEFINITIVA  y PROMEDIO aparecen como OBJECT**

In [ ]:
## **d. Revisamos nuevamente con _info()_ Aun la variable DEFINITIVA  y PROMEDIO aparecen como OBJECT
df_datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 121 entries, 0 to 120
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   CURSO         121 non-null    object
 1   DEFINITIVA    121 non-null    object
 2   PROGRAMA      121 non-null    object
 3   EDAD          121 non-null    int64 
 4   URBANO        121 non-null    object
 5   PESO          121 non-null    int64 
 6   ESTATURA      121 non-null    int64 
 7   SATISFACC     121 non-null    int64 
 8   SABER 11      121 non-null    int64 
 9   PROMEDIO      121 non-null    object
 10  SEXO          121 non-null    object
 11  ESTRATO       120 non-null    object
 12  Herramientas  121 non-null    object
dtypes: int64(5), object(8)
memory usage: 12.4+ KB


## **e. Mostrando analisis descriptivos de la información del DataFrame**

In [ ]:
## **e. Mostrando analisis descriptivos de la información del DataFrame**
df_datos.describe()

EDAD        PESO    ESTATURA  SATISFACC    SABER 11
count  121.000000  121.000000  121.000000  121.00000  121.000000
mean    18.553719   63.239669  167.933884    4.53719  305.429752
std      1.653030   12.935755   12.369812    0.63300   43.059228
min     16.000000   37.000000  100.000000    3.00000  204.000000
25%     18.000000   55.000000  163.000000    4.00000  275.000000
50%     18.000000   61.000000  169.000000    5.00000  305.000000
75%     19.000000   73.000000  175.000000    5.00000  341.000000
max     27.000000   97.000000  192.000000    5.00000  380.000000

#**39. Cambiando de OBJETO A NUMERICO en el DataFrame**

## **a. Usando el método _astype()_ hacemos el cambio de tipo de variables**

In [ ]:
## **a. Usando el método _astype()_ hacemos el cambio de tipo de variables**
df_datos["DEFINITIVA"]=df_datos["DEFINITIVA"].astype(np.float64)
df_datos["PROMEDIO"]=df_datos["PROMEDIO"].astype(np.float64)

## **b. Revisamos nuevamente con _info()_ Aun la variable DEFINITIVA  y PROMEDIO aparecen como OBJECT**

In [ ]:
## **b. Revisamos nuevamente con _info()_ Aun la variable DEFINITIVA  y PROMEDIO aparecen como OBJECT
df_datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 121 entries, 0 to 120
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   CURSO         121 non-null    object 
 1   DEFINITIVA    121 non-null    float64
 2   PROGRAMA      121 non-null    object 
 3   EDAD          121 non-null    int64  
 4   URBANO        121 non-null    object 
 5   PESO          121 non-null    int64  
 6   ESTATURA      121 non-null    int64  
 7   SATISFACC     121 non-null    int64  
 8   SABER 11      121 non-null    int64  
 9   PROMEDIO      121 non-null    float64
 10  SEXO          121 non-null    object 
 11  ESTRATO       120 non-null    object 
 12  Herramientas  121 non-null    object 
dtypes: float64(2), int64(5), object(6)
memory usage: 12.4+ KB


## **c. Mostrando analisis descriptivos de la información del DataFrame**
### _Al parecer hay algunos datos herrados en la Variable PROMEDIO__

In [ ]:
## **c. Mostrando analisis descriptivos de la información del DataFrame**
df_datos.describe()

DEFINITIVA        EDAD        PESO    ESTATURA  SATISFACC    SABER 11  \
count  121.000000  121.000000  121.000000  121.000000  121.00000  121.000000   
mean     3.773554   18.553719   63.239669  167.933884    4.53719  305.429752   
std      0.863980    1.653030   12.935755   12.369812    0.63300   43.059228   
min      1.100000   16.000000   37.000000  100.000000    3.00000  204.000000   
25%      3.500000   18.000000   55.000000  163.000000    4.00000  275.000000   
50%      4.000000   18.000000   61.000000  169.000000    5.00000  305.000000   
75%      4.500000   19.000000   73.000000  175.000000    5.00000  341.000000   
max      5.000000   27.000000   97.000000  192.000000    5.00000  380.000000   

         PROMEDIO  
count  121.000000  
mean     8.231901  
std     34.015054  
min      3.200000  
25%      3.800000  
50%      4.010000  
75%      4.300000  
max    373.000000

## **d. Con la función _len()_ tambien podemos explorar el tamaño del Data Frame**

In [ ]:
## **d. Con la función _len()_ tambien podemos explorar el tamaño del Data Frame**
len(df_datos) # numero de filas

121

## **e. La función _max_ y el atributo _index_ podemos visualizar el máximo en cada variable**

In [ ]:
## **e. La función _max_ y el atributo _index_ podemos visualizar el máximo en cada variable**
max(df_datos.index)

120

## **f. La función _min_ y el atributo _index_ podemos visualizar el mínimo**

In [ ]:
## **f. La función _min_ y el atributo _index_ podemos visualizar el mínimo**
min(df_datos.index)

0

## **g. La función _type_  la clase del DataFrame**

In [ ]:
## **g. La función _type_  la clase del DataFrame**
type(df_datos)

pandas.core.frame.DataFrame

## **h. La función _round_  podemos redondear en el DataFrame**

In [ ]:
## **h. La función _round_  podemos redondear en el DataFrame**
round(df_datos,1)

CURSO  DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  \
0    ESTADISTICA I         4.3     COMS    18     SI    55       164   
1    ESTADISTICA I         3.1     COMS    19     SI    65       165   
2    ESTADISTICA I         4.2     DERH    25     NO    56       170   
3    ESTADISTICA I         3.6     DERH    23     SI    65       170   
4    ESTADISTICA I         3.9     PSIC    19     SI    50       169   
5    ESTADISTICA I         1.1     PSIC    19     SI    68       165   
6    ESTADISTICA I         3.5    CIENP    19     SI    86       184   
7    ESTADISTICA I         4.2    CIENP    19     NO    53       163   
8    PROBABILIDAD          4.4     ININ    18     NO    56       158   
9    PROBABILIDAD          4.1     INMT    20     SI    72       179   
10   PROBABILIDAD          4.8     INNV    18     SI    64       175   
11   PROBABILIDAD          4.5     ININ    17     NO    75       177   
12   PROBABILIDAD          3.9     INQU    19     SI    61       158   
13   PROBABILIDAD          3.3     ININ    17     SI    57       164   
14   PROBABILIDAD          4.0     INMT    18     NO    50       168   
15   ESTADISTICA I         3.8     COMS    20     SI    87       174   
16   ESTADISTICA I         3.5     COMS    16     SI    54       169   
17   ESTADISTICA I         4.2     DERH    18     SI    80       181   
18   ESTADISTICA I         3.6     DERH    19     SI    60       182   
19   ESTADISTICA I         4.1     PSIC    18     SI    90       176   
20   ESTADISTICA I         2.5     PSIC    20     SI    50       157   
21   ESTADISTICA I         3.6    CIENP    19     SI    55       156   
22   ESTADISTICA I         4.5    CIENP    18     SI    74       183   
23   PROBABILIDAD          4.0     INST    18     NO    63       183   
24   PROBABILIDAD          5.0     INAM    17     SI    48       158   
25   ESTADISTICA I         3.6     DERH    18     SI    85       179   
26   ESTADISTICA I         4.0     PSIC    20     SI    60       175   
27   ESTADISTICA I         3.0     PSIC    17     NO    61       155   
28   ESTADISTICA I         4.0    CIENP    18     SI    68       185   
29   PROBABILIDAD          4.5     INMT    17     SI    85       175   
30   PROBABILIDAD          4.3     INBM    17     SI    45       155   
31   ESTADISTICA I         4.5     COMS    17     NO    60       162   
32   ESTADISTICA I         3.0     COMS    18     SI    65       169   
33   ESTADISTICA I         3.5     DERH    21     SI    62       100   
34   ESTADISTICA I         3.2     DERH    17     NO    60       170   
35   ESTADISTICA I         3.6     PSIC    18     SI    56       163   
36   ESTADISTICA I         4.0     PSIC    18     SI    75       190   
37   ESTADISTICA I         5.0    CIENP    19     SI    73       182   
38   ESTADISTICA I         3.8    CIENP    17     NO    73       175   
39   PROBABILIDAD          4.3     INMC    24     NO    90       176   
40   PROBABILIDAD          4.3     INMC    18     SI    47       167   
41   ESTADISTICA I         4.5     COMS    19     SI    50       165   
42   ESTADISTICA I         3.0     COMS    17     SI    70       173   
43   ESTADISTICA I         5.0     DERH    20     SI    63       160   
44   PROBABILIDAD          5.0     INCV    17     NO    37       148   
45   PROBABILIDAD          5.0     FINN    18     SI    58       165   
46   PROBABILIDAD          4.8     ININ    20     SI    73       163   
47   PROBABILIDAD          2.5     INMT    17     SI    64       180   
48   PROBABILIDAD          3.0     INNV    18     NO    80       175   
49   PROBABILIDAD          3.8     INMC    17     NO    70       170   
50   ESTADISTICA I         4.4     COMS    18     SI    60       165   
51   ESTADISTICA I         2.5     COMS    18     SI    77       180   
52   ESTADISTICA I         4.2     DERH    17     NO    59       169   
53   ESTADISTICA I         3.7     DERH    17     SI    40       164   
54   ESTADISTICA I         4.0     PSIC    18     SI 

#**40. Seleccionar una columna en el DataFrame para analizar los problemas**

In [ ]:
## **a. Revisando la columna PROMEDIO**
df_datos["PROMEDIO"]

0        4.26
1        4.11
2        3.70
3        4.10
4        4.00
5        3.70
6        4.48
7        3.70
8        3.80
9        3.90
10       4.69
11       4.69
12       3.77
13       3.77
14       4.30
15       3.80
16       4.48
17       3.79
18       4.00
19       3.90
20       3.90
21       3.86
22       4.31
23       4.02
24       4.22
25       3.82
26     373.00
27      37.00
28       4.01
29       3.78
30       4.12
31       3.80
32       3.81
33       3.90
34       4.10
35       4.30
36       4.03
37       4.08
38       4.26
39       3.70
40       4.20
41       3.90
42       4.03
43       4.00
44       3.87
45       4.22
46       3.73
47       3.88
48       3.80
49       4.00
50       4.68
51       4.10
52       4.01
53       4.33
54      42.60
55       4.10
56       3.90
57       4.10
58       4.30
59       4.20
60       4.20
61       4.46
62       3.60
63       4.20
64       4.01
65       4.57
66       4.30
67       3.79
68       4.01
69       4.60
70       4.40
71    

## **b. La Variable PROMEDIO tiene errores en la entrada**
|||
|:--:|:--:|
|26|     373.00|
|27|      37.00|
|54|      42.60|
|100|     35.00|
|112|     40.00|



## **c. Realizamos las correcciones pertinenetes en la Variable PROMEDIO**

In [ ]:
## **c. Realizamos las correcciones pertinenetes en la Variable PROMEDIO**
df_datos["PROMEDIO"]=df_datos["PROMEDIO"].replace(40.00,4.0)
df_datos["PROMEDIO"]=df_datos["PROMEDIO"].replace(35.00,3.5)
df_datos["PROMEDIO"]=df_datos["PROMEDIO"].replace(42.00,4.2)
df_datos["PROMEDIO"]=df_datos["PROMEDIO"].replace(37.00,3.7)
df_datos["PROMEDIO"]=df_datos["PROMEDIO"].replace(373.00,3.7)
df_datos["PROMEDIO"]=df_datos["PROMEDIO"].replace(42.60,4.3)
df_datos["PROMEDIO"]

0      4.26
1      4.11
2      3.70
3      4.10
4      4.00
5      3.70
6      4.48
7      3.70
8      3.80
9      3.90
10     4.69
11     4.69
12     3.77
13     3.77
14     4.30
15     3.80
16     4.48
17     3.79
18     4.00
19     3.90
20     3.90
21     3.86
22     4.31
23     4.02
24     4.22
25     3.82
26     3.70
27     3.70
28     4.01
29     3.78
30     4.12
31     3.80
32     3.81
33     3.90
34     4.10
35     4.30
36     4.03
37     4.08
38     4.26
39     3.70
40     4.20
41     3.90
42     4.03
43     4.00
44     3.87
45     4.22
46     3.73
47     3.88
48     3.80
49     4.00
50     4.68
51     4.10
52     4.01
53     4.33
54     4.30
55     4.10
56     3.90
57     4.10
58     4.30
59     4.20
60     4.20
61     4.46
62     3.60
63     4.20
64     4.01
65     4.57
66     4.30
67     3.79
68     4.01
69     4.60
70     4.40
71     4.10
72     4.50
73     3.95
74     4.20
75     4.40
76     4.30
77     3.83
78     4.69
79     3.80
80     3.50
81     3.90
82     3.80
83  

## **d. Nuevamente redondeamos y nos concentramos en la variable PROMEDIO**

In [ ]:
## **d. Nuevamente redondeamos y nos concentramos en la variable PROMEDIO**
round(df_datos["PROMEDIO"],1)

0      4.3
1      4.1
2      3.7
3      4.1
4      4.0
5      3.7
6      4.5
7      3.7
8      3.8
9      3.9
10     4.7
11     4.7
12     3.8
13     3.8
14     4.3
15     3.8
16     4.5
17     3.8
18     4.0
19     3.9
20     3.9
21     3.9
22     4.3
23     4.0
24     4.2
25     3.8
26     3.7
27     3.7
28     4.0
29     3.8
30     4.1
31     3.8
32     3.8
33     3.9
34     4.1
35     4.3
36     4.0
37     4.1
38     4.3
39     3.7
40     4.2
41     3.9
42     4.0
43     4.0
44     3.9
45     4.2
46     3.7
47     3.9
48     3.8
49     4.0
50     4.7
51     4.1
52     4.0
53     4.3
54     4.3
55     4.1
56     3.9
57     4.1
58     4.3
59     4.2
60     4.2
61     4.5
62     3.6
63     4.2
64     4.0
65     4.6
66     4.3
67     3.8
68     4.0
69     4.6
70     4.4
71     4.1
72     4.5
73     4.0
74     4.2
75     4.4
76     4.3
77     3.8
78     4.7
79     3.8
80     3.5
81     3.9
82     3.8
83     3.8
84     4.5
85     4.4
86     3.8
87     4.3
88     3.6
89     4.4
90     3.3

#**e. Seleccionar dos o mas columnas en el DataFrame**

In [ ]:
#**e. Seleccionar dos o mas columnas en el DataFrame**
df_datos[["PESO", "ESTATURA"]].head()

PESO  ESTATURA
0    55       164
1    65       165
2    56       170
3    65       170
4    50       169

#**f. Seleccionar dos o mas columnas en el DataFrame**

In [ ]:
#**f. Seleccionar dos o mas columnas en el DataFrame**
df_datos[["SEXO", "PESO", "ESTATURA"]].head()

SEXO  PESO  ESTATURA
0    F    55       164
1    F    65       165
2    M    56       170
3    F    65       170
4    F    50       169

# **41. Exportar dataframes limpios**

In [ ]:
# **41. Exportar dataframes limpios**
df_datos.to_csv('df_datos_LIMPIOS.csv',index=False)